src/primo/browzine-primo-adapter.js

```diff
@@ -1,78 +1,290 @@
+      browzineWebLink = data.browzineWebLink;
+    }
+
+    return browzineWebLink;
+  };
+
+  function getCoverImageUrl(scope, data, journal) {
+    var coverImageUrl = null;
+
+    if(isJournal(scope)) {
+      if(data && data.coverImageUrl) {
+        coverImageUrl = data.coverImageUrl;
+      }
+    }
+
+    if(isArticle(scope)) {
+      if(journal && journal.coverImageUrl) {
+        coverImageUrl = journal.coverImageUrl;
+      }
+    }
+
+    return coverImageUrl;
+  };
+
+  function getBrowZineEnabled(scope, data, journal) {
+    var browzineEnabled = false;
+
+    if(isJournal(scope)) {
+      if(data && data.browzineEnabled) {
+        browzineEnabled = data.browzineEnabled;
       }
-    }, function(error) {
-      console.log(error); //
-    });
-  }
-  vm.$doCheck = function(changes) {
-    if (vm.parentCtrl.selectedThumbnailLink) {
-      if (newThumbnail != '') {
-        vm.parentCtrl.selectedThumbnailLink.linkURL = newThumbnail;
+    }
+
+    if(isArticle(scope)) {
+      if(journal && journal.browzineEnabled) {
+        browzineEnabled = journal.browzineEnabled;
       }
     }
+
+    return browzineEnabled;
   };
-});
 
-app.component('prmSearchResultThumbnailContainerAfter', {
-  bindings: {
-    parentCtrl: '<'
-  },
-  controller: 'prmSearchResultThumbnailContainerAfterController'
-});
+  function buildTemplate(scope, browzineWebLink) {
+    var browzineWebLinkText = "";
+    var bookIcon = "https://assets.thirdiron.com/images/integrations/browzine_open_book_icon.png";
+
+    if(isJournal(scope)) {
+      browzineWebLinkText = browzine.journalBrowZineWebLinkText || browzine.primoJournalBrowZineWebLinkText || "View Journal Contents";
+    }
+
+    if(isArticle(scope)) {
+      browzineWebLinkText = browzine.articleBrowZineWebLinkText || browzine.primoArticleBrowZineWebLinkText || "View Issue Contents";
+    }
+
+    var template = "<div class='browzine'>" +
+                      "<a class='browzine-web-link' href='{browzineWebLink}' target='_blank' title='{browzineWebLinkText} in BrowZine'>" +
+                          "<img src='{bookIcon}' class='browzine-book-icon'/> {browzineWebLinkText}" +
+                      "</a>" +
+                   "</div>";
+
+    template = template.replace(/{browzineWebLink}/g, browzineWebLink);
+    template = template.replace(/{browzineWebLinkText}/g, browzineWebLinkText);
+    template = template.replace(/{bookIcon}/g, bookIcon);
+
+    return template;
+  };
+
+  function getElement(scope) {
+    return scope.$element;
+  };
+
+  function getElementParent(element) {
+    return element[0].offsetParent || element[0];
+  };
+
+  function getScope($scope) {
+    return $scope && $scope.$ctrl && $scope.$ctrl.parentCtrl ? $scope.$ctrl.parentCtrl : undefined;
+  };
+
+  function searchResult($scope) {
+    var scope = getScope($scope);
+
+    if(!shouldEnhance(scope)) {
+      return;
+    }
+
+    var endpoint = getEndpoint(scope);
+
+    var request = new XMLHttpRequest();
```

Comment:
For the time travelers to come, noting at this point in the timeline, jQuery was inaccessible in Primo's rollout of their Angular 1 web app for some reason. So it wasn't possible to do a `$.getJSON` or similar.